src/main/java/org/folio/innreach/domain/service/impl/BatchDomainEventProcessor.java

```diff
@@ -30,8 +30,12 @@ public <T> void process(List<DomainEvent<T>> batch, Consumer<DomainEvent<T>> rec
       var tenantId = tenantEventsEntry.getKey();
       var events = tenantEventsEntry.getValue();
 
-      executionService.runTenantScoped(tenantId,
-        () -> processTenantEvents(events, recordProcessor));
+      try {
+        executionService.runTenantScoped(tenantId,
+          () -> processTenantEvents(events, recordProcessor));
+      } catch (Exception ex) {
+        log.info("This operation is not permitted for system user [tenantId={}]", tenantId);
```

Comment:
This is a specific error message for a general exception type. It would be better to either handle a particular type of exception or change the log message and add more details regarding the cause of the error.